### DISCUSSION TITLE
Is there a way to detect changes to gitignored files without using `inputs`?

### DISCUSSION THREAD

**Discussion by thomasmattheussen** [vercel/turborepo#5745] [Help] [answered]
Is there a way to detect changes to gitignored files without using `inputs`?

I have a monorepo setup with about 40 packages of which 25 have the same `turbo.json`

I need to add `src/**` to the `inputs`, because it also contains some generated files that are gitignored. But this means, I have to explicitly add all dependencies it uses as input as well. And then I'd have to repeat it like 25 times.

```json
{
  "$schema": "https://turbo.build/schema.json",
  "extends": ["//"],
  "pipeline": {
    "build": {
      "inputs": ["src/**"],
      "outputs": ["lib/**", "lib-legacy/**", "lib-android/**"]
    },
    "compile": {
      "dependsOn": ["build"],
      "outputs": ["lib/**", "lib-legacy/**"]
    }
  }
}
```

**Comment by anthonyshew** [comment]
Short answer: No, `turbo` piggybacks on `git` for file changes so ignored files to `git` are ignored files to `turbo`. If you want them included, you'll need to mark them as `inputs`.

Longer answer: Does the codegen happen based on some other code in your repository? If that's the case, you would want to depend on that package that does the codegen (since a change there means a change in your gitignored files).

**Reply by anthonyshew** [reply, marked as answer]
We have much better answer for this today: [`$TURBO_DEFAULT$`](https://turbo.build/repo/docs/reference/configuration#turbo_default)
